### PR TITLE
epic(e2e): e2e tests improvements and extendtion 

### PR DESCRIPTION
**Pull request by ala-n**
Closes: #2267
Closes: #2266
Closes: #2265
Closes: #2263
Closes: #2264

### CODE REVIEW

File: e2e/setup/scenarios.screenshot.ts [hunk 1]
```diff
@@ -1,27 +1,42 @@
...
 };
 
-cucumber.defineRule('take a screenshot', async (world: TestEnv) => {
-  world.screenshots.push(await page.screenshot());
+cucumber.defineRule('take a screenshot', async ({screenshots}: TestEnv) => {
+  screenshots.push(await page.screenshot());
+});
+
+cucumber.defineRule('take a screenshot of a full page', async ({screenshots}: TestEnv) => {
+  screenshots.push(await page.screenshot({fullPage: true}));
```
Comment (dshovchko) [inline e2e/setup/scenarios.screenshot.ts:15]: Can we save screenshots in WEBP format?

Comment (fshovchko) [reply e2e/setup/scenarios.screenshot.ts:15]: > Can we save screenshots in WEBP format?

Not yet, but there's a ticket opened for it https://github.com/exadel-inc/esl/issues/2393

File: e2e/setup/scenarios.screenshot.ts [hunk 2]
```diff
@@ -1,27 +1,42 @@
...
+  if (!element) throw new Error('E2E: there is no any element, make sure you have "Find the element by selector" before');
+
+  await element.scrollIntoView();
+  screenshots.push(await element.screenshot());
 });
 
-cucumber.defineRule('check if the screenshot is exactly equal to the snapshoted version', (world: TestEnv) => {
-  if (!world.screenshots.length) throw new Error('E2E: there is no any screenshot, make sure you have "Take a screenshot" before');
-  const image = world.screenshots.pop();
+cucumber.defineRule('check if the screenshot is exactly equal to the snapshoted version', ({screenshots}: TestEnv) => {
```
Comment (Natalie-Smirnova) [inline e2e/setup/scenarios.screenshot.ts:32]: cosmetic: snapshotted

**Review by Natalie-Smirnova — APPROVED** [review]

File: e2e/setup/scenarios.screenshot.ts [hunk 3]
```diff
@@ -1,27 +1,42 @@
...
+cucumber.defineRule('take a screenshot of a full page', async ({screenshots}: TestEnv) => {
+  screenshots.push(await page.screenshot({fullPage: true}));
 });
 
-cucumber.defineRule('take a screenshot of a full page', async (world: TestEnv) => {
-  world.screenshots.push(await page.screenshot({fullPage: true}));
+cucumber.defineRule('scroll to the element', async ({elements}: TestEnv) => {
+  const element = elements[0];
+  if (!element) throw new Error('E2E: there is no any element, make sure you have "Find the element by selector" before');
+  await element.scrollIntoView();
```
[the remark this answers is not in the document]
Comment (ala-n) [reply e2e/setup/scenarios.screenshot.ts:21]: Yes and now, scrollIntoView is not animated but all operations trough ElementHandler (that BrowserDriver gives us) are asynchronous

File: e2e/setup/element/scenarios.basic.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import {cucumber} from '../../transformer/gherkin';
+import type {TestEnv} from '../scenarios.env';
+
+cucumber.defineRule(/check if the elements? is present/, async ({elements}: TestEnv) => {
```
[the remark this answers is not in the document]
Comment (ala-n) [reply e2e/setup/element/scenarios.basic.ts:4]: Yep, cucumber accept rules as a string or RegExp

File: e2e/setup/scenarios.screenshot.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (fshovchko) [reply e2e/setup/scenarios.screenshot.ts:21]: Yep, it's promisified: https://pptr.dev/api/puppeteer.elementhandle.scrollintoview

[the remark this answers is not in the document]
Comment (fshovchko) [reply e2e/setup/scenarios.screenshot.ts:21]: > Do they implement promisify scrollIntoView?

Yep, it's promisified: https://pptr.dev/api/puppeteer.elementhandle.scrollintoview

File: e2e/setup/scenarios.page.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import {cucumber} from '../transformer/gherkin';
+
+const port = process.env.PORT || 3007;
+
+cucumber.defineRule('a page {string}', async (env: unknown, path: string) => {
+  const url = new URL(path, `http://localhost:${port}/`);
```
[the remark this answers is not in the document]
Comment (ala-n) [reply e2e/setup/scenarios.page.ts:6]: Nope
image
Everything is fine

**Pull request merged**